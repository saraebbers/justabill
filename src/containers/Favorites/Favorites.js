import React from 'react';
import { connect } from 'react-redux';
import '../../index.scss';
import PropTypes from 'prop-types';
import Card from '../Card/Card';


export const Favorites = (props) => {

  const { billArray, favoritesArray } = props

  let information;

  if(!favoritesArray.length) {
    information = <div className='container'><p className='msg'>There are no selected favorites to display</p></div>
  } else {
    information = billArray.map(billItem => {
      return billItem.bills.map(bill => { 
        if(favoritesArray.includes(bill.id)) {
          return <Card {...bill} key={bill.id} />
        } else {
          return null
        }
      })
    })
  }

  return(
    <div className='favorite-pg'>
      <section className='favorite-container'>
        <div className='fav-msg'>
          <h3>
            Below are your selected favorites.
          </h3>
        </div>
        <div className='cardContainer'>
          { information }
        </div>
      </section>
    </div>
  )
}

export const mapStateToProps = (state) => ({
  billArray: state.bills,
  favoritesArray: state.favorites
})


Favorites.propTypes = {
  billArray: PropTypes.array.isRequired,
  favoritesArray: PropTypes.array.isRequired
}

export default connect(mapStateToProps)(Favorites);

  