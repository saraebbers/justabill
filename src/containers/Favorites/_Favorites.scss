.favorite-container {
  height: 100vh;
  background: url(./assets/congressFloor.jpg)
    top center;
  background-size: cover;
}
