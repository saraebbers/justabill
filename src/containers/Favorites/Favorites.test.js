import React from 'react';
import { shallow } from 'enzyme';
import { mapStateToProps, mapDispatchToProps, Favorites } from './Favorites';


describe('Favorites', () => {
  let mockBillArray;
  let mockFavoritesArray;

  beforeEach(() => {
    mockBillArray=[{congress: 116, bills: [{id: 'hr-123', name:'sara', somethingelse: 'xxx'}, {id: 'hr234', name:'David', somethingelse: 'xxx'}]}]
    mockFavoritesArray=['hr-45123', 'hr-123']
  })

  describe('Favorites container', () => {

    it('should match the snapshot if there are favorites', () => {
      const wrapper = shallow(<Favorites favoritesArray={mockFavoritesArray} billArray={mockBillArray}  />)
      expect(wrapper).toMatchSnapshot()
    })

    it('should match the snapshot if there are no favorites', () => {
      mockFavoritesArray=[]
      const wrapper = shallow(<Favorites favoritesArray={mockFavoritesArray} billArray={mockBillArray}  />)
      expect(wrapper).toMatchSnapshot()
    })

  })

  describe('mapStateToProps', () => {
    it('should return a props object with the correct information from state parsed', () => {
      const mockState = {
        bills: mockBillArray,
        errorMessage: 'this is an error message',
        favorites: mockFavoritesArray,
        somethingElse: 'something else'
      }
      const expected = {
        billArray: mockBillArray,
        favoritesArray: mockFavoritesArray
      }
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })
  })

})
