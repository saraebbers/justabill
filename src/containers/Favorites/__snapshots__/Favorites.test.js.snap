// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites Favorites container should match the snapshot if there are favorites 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Favorites
    billArray={
      Array [
        Object {
          "bills": Array [
            Object {
              "id": "hr-123",
              "name": "sara",
              "somethingelse": "xxx",
            },
            Object {
              "id": "hr234",
              "name": "David",
              "somethingelse": "xxx",
            },
          ],
          "congress": 116,
        },
      ]
    }
    favoritesArray={
      Array [
        "hr-45123",
        "hr-123",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <section
        className="favorite-container"
      >
        <div
          className="fav-msg"
        >
          <h3>
            Below are your selected favorites.
          </h3>
        </div>
        <div
          className="cardContainer"
        >
          <withRouter(Connect(Card))
            id="hr-123"
            name="sara"
            somethingelse="xxx"
          />
        </div>
      </section>,
      "className": "favorite-pg",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="fav-msg"
          >
            <h3>
              Below are your selected favorites.
            </h3>
          </div>,
          <div
            className="cardContainer"
          >
            <withRouter(Connect(Card))
              id="hr-123"
              name="sara"
              somethingelse="xxx"
            />
          </div>,
        ],
        "className": "favorite-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h3>
              Below are your selected favorites.
            </h3>,
            "className": "fav-msg",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Below are your selected favorites.",
            },
            "ref": null,
            "rendered": "Below are your selected favorites.",
            "type": "h3",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              Array [
                <withRouter(Connect(Card))
                  id="hr-123"
                  name="sara"
                  somethingelse="xxx"
                />,
                null,
              ],
            ],
            "className": "cardContainer",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "hr-123",
              "nodeType": "function",
              "props": Object {
                "id": "hr-123",
                "name": "sara",
                "somethingelse": "xxx",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            null,
          ],
          "type": "div",
        },
      ],
      "type": "section",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <section
          className="favorite-container"
        >
          <div
            className="fav-msg"
          >
            <h3>
              Below are your selected favorites.
            </h3>
          </div>
          <div
            className="cardContainer"
          >
            <withRouter(Connect(Card))
              id="hr-123"
              name="sara"
              somethingelse="xxx"
            />
          </div>
        </section>,
        "className": "favorite-pg",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="fav-msg"
            >
              <h3>
                Below are your selected favorites.
              </h3>
            </div>,
            <div
              className="cardContainer"
            >
              <withRouter(Connect(Card))
                id="hr-123"
                name="sara"
                somethingelse="xxx"
              />
            </div>,
          ],
          "className": "favorite-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h3>
                Below are your selected favorites.
              </h3>,
              "className": "fav-msg",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Below are your selected favorites.",
              },
              "ref": null,
              "rendered": "Below are your selected favorites.",
              "type": "h3",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                Array [
                  <withRouter(Connect(Card))
                    id="hr-123"
                    name="sara"
                    somethingelse="xxx"
                  />,
                  null,
                ],
              ],
              "className": "cardContainer",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "hr-123",
                "nodeType": "function",
                "props": Object {
                  "id": "hr-123",
                  "name": "sara",
                  "somethingelse": "xxx",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              null,
            ],
            "type": "div",
          },
        ],
        "type": "section",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Favorites Favorites container should match the snapshot if there are no favorites 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Favorites
    billArray={
      Array [
        Object {
          "bills": Array [
            Object {
              "id": "hr-123",
              "name": "sara",
              "somethingelse": "xxx",
            },
            Object {
              "id": "hr234",
              "name": "David",
              "somethingelse": "xxx",
            },
          ],
          "congress": 116,
        },
      ]
    }
    favoritesArray={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <section
        className="favorite-container"
      >
        <div
          className="fav-msg"
        >
          <h3>
            Below are your selected favorites.
          </h3>
        </div>
        <div
          className="cardContainer"
        >
          <div
            className="container"
          >
            <p
              className="msg"
            >
              There are no selected favorites to display
            </p>
          </div>
        </div>
      </section>,
      "className": "favorite-pg",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="fav-msg"
          >
            <h3>
              Below are your selected favorites.
            </h3>
          </div>,
          <div
            className="cardContainer"
          >
            <div
              className="container"
            >
              <p
                className="msg"
              >
                There are no selected favorites to display
              </p>
            </div>
          </div>,
        ],
        "className": "favorite-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h3>
              Below are your selected favorites.
            </h3>,
            "className": "fav-msg",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Below are your selected favorites.",
            },
            "ref": null,
            "rendered": "Below are your selected favorites.",
            "type": "h3",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="container"
            >
              <p
                className="msg"
              >
                There are no selected favorites to display
              </p>
            </div>,
            "className": "cardContainer",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <p
                className="msg"
              >
                There are no selected favorites to display
              </p>,
              "className": "container",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "There are no selected favorites to display",
                "className": "msg",
              },
              "ref": null,
              "rendered": "There are no selected favorites to display",
              "type": "p",
            },
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "section",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <section
          className="favorite-container"
        >
          <div
            className="fav-msg"
          >
            <h3>
              Below are your selected favorites.
            </h3>
          </div>
          <div
            className="cardContainer"
          >
            <div
              className="container"
            >
              <p
                className="msg"
              >
                There are no selected favorites to display
              </p>
            </div>
          </div>
        </section>,
        "className": "favorite-pg",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="fav-msg"
            >
              <h3>
                Below are your selected favorites.
              </h3>
            </div>,
            <div
              className="cardContainer"
            >
              <div
                className="container"
              >
                <p
                  className="msg"
                >
                  There are no selected favorites to display
                </p>
              </div>
            </div>,
          ],
          "className": "favorite-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h3>
                Below are your selected favorites.
              </h3>,
              "className": "fav-msg",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Below are your selected favorites.",
              },
              "ref": null,
              "rendered": "Below are your selected favorites.",
              "type": "h3",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="container"
              >
                <p
                  className="msg"
                >
                  There are no selected favorites to display
                </p>
              </div>,
              "className": "cardContainer",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <p
                  className="msg"
                >
                  There are no selected favorites to display
                </p>,
                "className": "container",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "There are no selected favorites to display",
                  "className": "msg",
                },
                "ref": null,
                "rendered": "There are no selected favorites to display",
                "type": "p",
              },
              "type": "div",
            },
            "type": "div",
          },
        ],
        "type": "section",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
